compiler/ast.nim

```diff
@@ -1279,6 +1289,8 @@ proc copyType*(t: PType, owner: PSym, keepId: bool): PType =
     when debugIds: registerId(result)
   result.sym = t.sym          # backend-info should not be copied
 
+proc exactReplica*(t: PType): PType = copyType(t, t.owner, true)
```

Comment:
General remark: This is a bad idea. Types can be mutated in the codegen ``field.loc`` etc and the write only affects one "replica" and not the other, causing subtle codegen bugs.